rest-api/dao/participant_dao.py

```diff
@@ -114,6 +114,15 @@ def _do_update(self, session, obj, existing_obj):
                             else None)
       need_new_summary = True
 
+    # If the provider link changes, update the HPO ID on the participant and its summary.
+    if obj.hpoId is None:
+      obj.hpoId = existing_obj.hpoId
+    if obj.providerLink != existing_obj.providerLink and obj.providerLink != 'null':
```

Comment:
I think it should be "obj.providerLink is not None"... not != 'null'.

Reply:
added a check for hpo_id being 0 as well.